shared-libs/outbound/src/outbound.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Outbound is a service that creates and sends REST requests to external services based on data
+ * generated in our system, and configurations the define mappings, external service location and
+ * authentication.
+ *
+ * See: https://github.com/medic/medic-docs/blob/master/configuration/outbound.md
+ *
+ * This library deals with the conversion of the record into a REST request, and sending it to its
+ * destination. Background management of retries and so on is done in schedules, and the initial
+ * requester of an outbound push is the outbound transition.
+ */
+const objectPath = require('object-path');
+const urlJoin = require('url-join');
```

Comment:
Can we just use the node `path` module here?

Reply:
...Yes? Maybe? So this came up initially, and [back then I wrote this: (this link should work but doesn't deep link right, sigh github)](https://github.com/medic/cht-core/pull/5686/files#r286925613)

> I'm not sure if the url library can help here (I did look, and `path` is just for disk paths not for uris):

But that doesn't seem true? After playing around with it, including rolling my node back to 8.x, and it seems to work fine. I can't remember that far back, perhaps back then I just looked at the documentation and concluded this (the docs are very OS path focused) as opposed to actually trying it? 

I'll change it to use path, and hope that I didn't catch some really important edge case back there that I'm not catching now.

Reply:
U N I T   T E S T S

```
1) outbound shared library
       push
         should push on minimal configuration:

      AssertionError: expected 'http:/test/foo' to equal 'http://test/foo'
      + expected - actual

      -http:/test/foo
      +http://test/foo

    at /home/scdf/Code/Medic/cht-6306/shared-libs/outbound/test/outbound.spec.js:255:18
```

So yeah, `path` *is* only for file systems (it's reading it like "C:/blah").

Reply:
Ugh. Good catch!

Reply:
Bro I'm just chuffed unit tests I wrote actually caught a bug :open_mouth: